### PR TITLE
chore(nimbus-ui): hook up storybook publishing on build

### LINKED ISSUE #3427: Automate pushing Storybook builds up to gh-pages
FxA automatically deploys Storybook builds to [https://mozilla-fxa.github.io/storybooks/](https://mozilla-fxa.github.io/storybooks/) which has been very useful to send to the design team and others to share the state of the Settings project and easily receive feedback. Let's do the same for the stories in the Nimbus project.

See https://jira.mozilla.com/browse/FXA-1542 for the original issue in FxA. [https://github.com/storybookjs/storybook-deployer](https://github.com/storybookjs/storybook-deployer) is mentioned in issue discussion which might be sufficient enough and a better choice for this project. Otherwise, we may need a modified version of [https://github.com/mozilla/fxa/blob/main/_scripts/build-storybooks-indexes.js](https://github.com/mozilla/fxa/blob/main/_scripts/build-storybooks-indexes.js) and [https://github.com/mozilla/fxa/blob/main/_scripts/build-storybooks.sh](https://github.com/mozilla/fxa/blob/main/_scripts/build-storybooks.sh).

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/EXP-281)
┆Issue Number: EXP-281

### PR DESCRIPTION
**Pull request by lmorchard**
Because

* We want to publish Storybook builds to demo and preview component work
  in nimbus-ui

This commit

* Integrates [mozilla-fxa/storybook-gcp-publisher](github.com/mozilla-fxa/storybook-gcp-publisher) into the CircleCI flow
  so that Storybooks are built and published to a Google Cloud static
  site for every test run

EXP-281
fixes #3427

### CODE REVIEW

**Comment by lmorchard** [comment]
Opening as a draft initially to see if my CircleCI plumbing works. (It probably won't on the first try)

**Comment by lmorchard** [comment]
Hooray, got this passing after tweaking some GCP permissions!

To see the storybook build for this PR, click "Show all checks" under "All checks have passed" and check out the "Details" link for "storybooks: pull request"

https://storage.googleapis.com/mozilla-storybooks-experimenter/commits/9dbfc7c70f6fd6588e8f496f7300eed78525e54a/index.html

**Comment by lmorchard** [comment]
I also just tried spreading the Owner role on the Google Cloud project where this site lives. Hopefully that reduces the bus factor a bit. The Github side of things just requires a Personal Access Token from anyone with commit access to the repos, which can be created with access to just post status checks.

I already created all the credentials and env vars, but in case anything goes wrong other folks should have the keys.

**Review by LZoog — APPROVED** [review]
@lmorchard Thanks so much for your work here! (you might bump this task up to a 5 or 8 before merging to better reflect the work done BTW, I made it a "3" since I assumed you'd copy and paste from FxA but this is such a cleaner solution.)

Will you add [your comment](https://github.com/mozilla/experimenter/pull/3579#issuecomment-701004921) to the nimbus-ui [Storybook docs](https://github.com/mozilla/experimenter/tree/main/app/experimenter/nimbus-ui#storybook)? 🙂

**Comment by lmorchard** [comment]
Rebased, added a bit to the docs about Storybook publishing

**Pull request merged**
